### PR TITLE
Fix bug in tutorial agent giving money to itself

### PR DESCRIPTION
**Pull request by houssam7737**
Discussed this with @jackiekazil and @rht 
This is more accurate and makes the dynamics more precise.

Rerun all cells to make sure all outputs/graphs are updated.

I will modify the code in the examples repo next since it has the same issue.

### CODE REVIEW

**Comment by houssam7737** [comment]
pre-commit.ci autofix

**Comment by houssam7737** [comment]
Mmmh codespell is running on the "image" portion of the notebook. Should we force merge given that the codespell does not make much sense in this case?

File: docs/tutorials/intro_tutorial.ipynb [hunk 1]
```diff
@@ -710,9 +890,30 @@
...
...1dTXW99NJL0dHRER/5yEfii1/8Yvz617+u2yx1f1fbWjt27FicPn16zG9XnT17dvzyl7+s01RQHUNDQ3HPPffEZz7zmbj66qvrPU7D+cUvfhFLliyJt956Ky699NLYtm1bfPKTn6z3WA3nkUceieeeey6effbZeo/SsBYvXhxbt26NK6+8Mg4fPhw9PT3xx3/8x/HCCy9Ea2tr+jwNHx/QyLq6uuKFF16o63O3jezKK6+MAwcORH9/f/zXf/1XrFu3Lnbt2iVAquiVV16Jr371q9HX1xdTp06t9zgNa9WqVSP/e8GCBbF48eK44oor4j/+4z/q8jRiw8fHzJkzY9KkSXH06NFRx48ePRrt7e11mgou3F133RWPPfZY7N69O+bOnVvvcRrSlClT4mMf+1hERFx77bXx7LPPxj/8wz/ED37wgzpP1jj2798fr7/+enz6058eOXb69OnYvXt3fPe7341SqRSTJk2q44SN6bLLLouPf/zjcfDgwbrcf8O/5mPKlClx7bXXxs6dO0eODQ0Nxc6dOz13y4RUqVTirrvuim3btsVPf/rTmD9/fr1HumgMDQ1FqVSq9xgN5ZZbbolf/OIXceDAgZGPRYsWxRe/+MU4cOCA8KiRkydPxqFDh2LOnDl1uf+Gf+QjIqK7uzvWrVsXixYtiuuvvz42bdoUg4ODcccdd9R7tIZy8uTJURX98ssvx4EDB2L69Okxb968Ok7WWLq6uqK3tzceffTRaG1tjSNHjkRExLRp0+KSSy6p83SNY/369bFq1aqYN29enDhxInp7e+Ppp5+OJ554ot6jNZTW1tYxr1f60Ic+FDNmzPA6pir6+te/HqtXr44rrrgiXnvttbj//vtj0qRJceutt9ZlnosiPr7whS/EG2+8Effdd18cOXIkrrnmmtixY8eYF6FyYfbt2xc33XTTyO3u7u6IiFi3bl1s3bq1TlM1ni1btkRExNKlS0cdf/jhh+P222/PH6hBvf766/GlL30pDh8+HNOmTYsFCxbEE088EcuXL6/3aDBur776atx6663x5ptvxuWXXx433HBD7N27Ny6//PK6zHNR/J4PAOCDo+Ff8wEAfLCIDwAglfgAAFKJDwAglfgAAFKJDwAglfgAAFKJDwAglfgAAFKJDwAglfgAAFKJDwAg1f8DvnKetViEmGQAAAAASUVORK5CYII=",
```
Comment (jackiekazil) [inline docs/tutorials/intro_tutorial.ipynb:908]: This is the line with offending code spell. I think we should ignore and merge... this is not intended for codespell.

**Comment by jackiekazil** [comment]
This looks to me.

**Pull request merged**